File: redsheet-portal/src/app/database-models/audit-record-detail.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { AuditRecord,  } from './'
export class AuditRecordDetail {
    constructor(auditRecordDetail?: AuditRecordDetail )  {
        let properties = [ "auditRecordDetailId", "columnName", "newValue", "oldValue", "referenceTableName", "auditRecordId",];
        for (let property of properties)
            if (auditRecordDetail && auditRecordDetail[property])
                this[property] = auditRecordDetail[property];
    }
 
	auditRecordDetailId : number =   0 ;
 
    @required()
    @maxLength(50)
	columnName : string =   undefined;
 
	newValue : string =   undefined;
 
	oldValue : string =   undefined;
 
    @required()
    @maxLength(50)
	referenceTableName : string =   undefined;
 
    @range(0,2147483647)
	auditRecordId : number =   undefined;
	auditRecord : AuditRecord  ;


}

File: RedSheet.DbEntities/Models/AuditRecordDetail.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("AuditRecordDetails", Schema = "dbo")]
	[Serializable]
	public partial class AuditRecordDetail  
	{
	///<summary>
    ///Record Details
    ///</summary> 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int AuditRecordDetailId { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string ColumnName { get; set; } 

	public string NewValue { get; set; } 

	public string OldValue { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string ReferenceTableName { get; set; } 
 
	[RelationshipTableAttribue("AuditRecords","dbo")]
	//Relationships
	public int AuditRecordId { get; set; } 
 
	[ForeignKey("AuditRecordId")]
	public AuditRecord AuditRecord { get; set; } 

	public AuditRecordDetail()
	{
    }

    }
}

